packages/medusa/src/services/cart.ts

```diff
@@ -1462,9 +1462,18 @@ class CartService extends TransactionBaseService {
             relations: ["rule", "rule.conditions", "regions"],
           })
 
-        await this.discountService_
-          .withTransaction(transactionManager)
-          .validateDiscountForCartOrThrow(cart, discounts)
+        try {
+          await this.discountService_
+            .withTransaction(transactionManager)
+            .validateDiscountForCartOrThrow(cart, discounts)
+        } catch (error) {
+          if (
+            error?.code === MedusaError.Codes.CUSTOMER_DISCOUNT_LIMIT_REACHED
```

Comment:
todo: This error code seems too specific. Can we change this to `NOT_ALLOWED` as with the rest of the validation in `validateDiscountForCartOrThrow`?